AnimationSmoothness/explainer.md

```diff
@@ -0,0 +1,175 @@
+
+### Option 3: Event Listener 
+
+Adding an event listener for frame rate changes would alert developers about large drops in frame rate. Since it would not be necessary to know if the rate drops by a frame or two. Instead, the developer could set the event listener to alert when the frame rate drops by n. Or, similarly to the long task API's duration threshold, the developer could set a min and max fps. The event listener would fire only if the FPS is above the max or below the min. 
+
+`window.addEventListener("frameratechange", (event) =>{doSomething();}) `
+
+### Option 4: Performance Observer
+
+This option works similarly to both [LoAF API](https://github.com/w3c/long-animation-frames) and the [Paint Timing API](https://www.w3.org/TR/paint-timing/), which both use the performance observer and follow a pattern that developers expect to use when improving performance. When observing long animation frames, developers can specify the entry types they want to the performance observer to processes. Like the performance observer reports which animation frames are too long, the event listener would send an alert when the frame rate drops by a certain amount. The two APIs differ in the amount of information given. The LoAF API can give more specific metrics for long animations, while event listeners provide a more general way of monitoring frame rate. 
+
+```javascript
+function perfObserver(list, observer) { 
+ list.getEntries().forEach((entry) => { 
+    if (entry.entryType === "frameSmoothness") { 
+      console.log(${entry.name}'s startTime: ${entry.startTime}); 
+      }
+     }); 
+} 
+const observer = new PerformanceObserver(perfObserver); 
+observer.observe({ entryTypes: ["frameSmoothness"] });
+
+```
+
+## Alternatives Considered
+For the event listener scenario, it was determined that using granularity would not give a useful measure of frame info due to lack of detail. The granularity was modeled after the compute pressure API.
+`window.addEventListener("frameratechange", (event) =>{doSomething();})`
+
+## Concerns/Open Questions
+1. The user-perceived smoothness is influenced by both the main thread and the compositor thread. Accurate measurement of frame rates must account for both. Since the compositor thread operates independently of the main thread, it can be difficult to get its frame rate data. However, an accurate frame rate measurements needs to take into account both measurements.
+2. Similar to the abandoned [Frame Timing interface](https://wicg.github.io/frame-timing/#introduction). We are currently gathering historical context on how this relates and why it is no longer being pursued.
+3. Questions to Consider:
+	* Should content missing from the compositor frame due to delayed tile rasterization be tracked? 
```

Comment:
Some folks here have started calling this concept frame "contentfulness", to differentiate raw frame "smoothness".

It can include more than just tile raster (such as image decode), including the the actual design of page contents (such as video bitrate / buffering, or real time video game latency, or typical web page loading related layout shifts, images, ads, fonts...).  Users seem more sensitive to that stuff than literal graphics pipeline performance (if trying to judge "quality of experience").